packages/clay-autocomplete/src/ClayAutocomplete.js

```diff
@@ -109,10 +111,10 @@ class ClayAutocomplete extends ClayComponent {
 	 * @return {Boolean} If the event has been prevented or not.
 	 */
 	_handleOnInput(event) {
-		const query = event.target.value;
+		this._query = event.target.value;
 
 		if (this.enableAutocomplete) {
-			this._handleFilteredItems(event, query);
+			this.refs.dataProvider.updateData(this._query);
```

Comment:
hey @4lejandrito, we have to be careful for use cases that do not need a query in the URL or that do not need to make requests to each user input, in this way will always request a request unnecessarily.